File: frontend/src/services/fieldService.ts
import { Field, FieldCreateRequest } from '../types/Field'
import { api } from './api'

export const fieldService = {
  async getAllFields(): Promise<Field[]> {
    const response = await api.get<Field[]>('/fields')
    return response.data
  },

  async getFieldById(id: number): Promise<Field> {
    const response = await api.get<Field>(`/fields/${id}`)
    return response.data
  },

  async createField(field: FieldCreateRequest): Promise<Field> {
    const response = await api.post<Field>('/fields', field)
    return response.data
  },

  async updateField(id: number, field: FieldCreateRequest): Promise<Field> {
    const response = await api.put<Field>(`/fields/${id}`, field)
    return response.data
  },

  async deleteField(id: number): Promise<void> {
    await api.delete(`/fields/${id}`)
  },
}

File: agriculture-field-service/src/main/java/com/omstu/agriculturefield/field/service/FieldService.java
package com.omstu.agriculturefield.field.service;

import com.omstu.agriculturefield.field.dto.AgriculturalFieldRequest;
import com.omstu.agriculturefield.field.dto.AgriculturalFieldResponse;

import java.util.List;

public interface FieldService {
    AgriculturalFieldResponse createField(AgriculturalFieldRequest request, Long farmId);
    AgriculturalFieldResponse updateField(Long id, AgriculturalFieldRequest request, Long farmId);
    AgriculturalFieldResponse getFieldById(Long id, Long farmId);
    List<AgriculturalFieldResponse> getAllFields(Long farmId);
    void deleteField(Long id, Long farmId);
}
